compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs

```diff
@@ -31,7 +31,7 @@ pub enum TypeAnnotationNeeded {
     /// ```
     E0282,
     /// An implementation cannot be chosen unambiguously because of lack of information.
-    /// ```compile_fail,E0283
+    /// ```compile_fail
```

Comment:
We should make sure that `E0283` is still emitted here. If this doesn't emit it, it should be replaced with something else that does (maybe you'll find something in the UI tests)

Reply:
It seems that in some cases `compile_fail, E****` probably does not work well for some error codes.
https://github.com/rust-lang/rust/pull/99324#discussion_r985158351
More details will need to be looked into.

Reply:
I just remembered that basically there are several error codes here that get emitted in slightly different cases (:/), so removing it is probably fine.